tests/test_system.py

```diff
@@ -0,0 +1,222 @@
+    c.expect('Hit enter.*')
+    c.sendline('')
+    if os.name == 'nt':
+        c.logfile.close()
+
+
+class TestSystem:
+
+    @classmethod
+    def setupClass(cls):
+        doInitalConfig('Mäf', 'müfo')
+
+    def test_login(self):
+        c = p_spawn(PY_PATH + ' ' + pyloadpath)
+        if os.name != 'nt':
+            c.logfile = sys.stdout
+        else:
+            lgfile = open('login_logfile.txt', 'w')
+            c.logfile = lgfile
+        # Wait pyload is running
+        c.expect('ADDON ClickNLoad: Proxy listening.*')
+        # Login via API
+        r = requests.Session()
+        u = r.post(api_url % "login", data={"username": "Mäf", "password": "müfo"})
+        self.key = u.json()
+        assert self.key is not False
+        r.close()
+        # Login Via Web Interface
+        r = requests.Session()
+        u = r.post(url % "login", data={"username": "Mäf",
+                                        "password": "müfo",
+                                        "submit": "login",
+                                        "do": "login"})
+        content = u.text
+        assert "Incorrect username/email or password." not in content
+        assert "Logout" in content
+        r.close()
+        if os.name == 'nt':
+            c.logfile.close()
+
+    def test_change_password(self):
+        c = p_spawn(PY_PATH + ' ' + pyloadpath + ' -u')
+        if os.name != 'nt':
+            c.logfile = sys.stdout
+        else:
+            lgfile = open('password_logfile.txt', 'w')
+            c.logfile = lgfile
+        c.expect('.*/2/3/4:')
+        c.sendline('1')
+        c.expect('Username.*')
+        c.sendline('Mäf')
+        c.expect('Password.*')
+        c.sendline('Külu')
+        c.expect('Password.*again.*')
+        c.sendline('Külu')
+        c.expect('.*/2/3/4:')
+        c.sendline('4')
+        if os.name == 'nt':
+            c.logfile.close()
+        c = p_spawn(PY_PATH + ' ' + pyloadpath)
+        if os.name != 'nt':
+            c.logfile = sys.stdout
+        else:
+            lgfile = open('pyload_logfile.txt', 'w')
+            c.logfile = lgfile
+        # Wait pyload is running
+        c.expect('ADDON ClickNLoad: Proxy listening.*')
+        # Login Via Web Interface
+        r = requests.Session()
+        u = r.post(url % "login", data={"username": "Mäf",
+                                        "password": "Külu",
+                                        "submit": "login",
+                                        "do": "login"})
+        content = u.text
+        assert "Logout" in content
+        # Change Password in UI
+        u = r.post(url % "json/change_password", data={"user_login": "Mäf",
+                                                       "login_current_password": "Külu",
+                                                       "login_new_password": "müfo",
+                                                       "login_new_password2": "müfo"})
+        assert u.status_code == 200
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=5985155)